cognite/neat/_data_model/validation/dms/_reverse_connection_validators.py

```diff
@@ -0,0 +1,346 @@
+
+
+class BidirectionalConnectionMisconfigured(DataModelValidator):
+    """This validator checks bidirectional connections to ensure reverse and direct connection pairs
+    are properly configured.
+
+    A bidirectional connection consists of:
+    - A reverse connection property in a target view, pointing to a source view through a direct connection property
+    - A corresponding direct connection property in a source view that points back to the target view
+
+    Example:
+        View A has property "relatedB" (reverse) → through View B property "relatedA" (direct)
+        View B has property "relatedA" (direct) → points back to View A
+
+    Validation checks:
+        1. Source view and property exist
+        2. Property is a direct connection type
+        3. Container mapping is correct
+        4. Direct connection points back to correct target
+    """
+
+    code = "NEAT-DMS-004"
+
+    def run(self) -> list[ConsistencyError | Recommendation]:
+        """Run validation and return list of issues found."""
+        issues: list[ConsistencyError | Recommendation] = []
+
+        for (target_view_ref, reverse_prop_name), (
+            source_view_ref,
+            through,
+        ) in self.local_resources.reverse_to_direct_mapping.items():
+            through = self._normalize_through_reference(source_view_ref, through)
+            context = ReverseConnectionContext(target_view_ref, reverse_prop_name, source_view_ref, through)
+            issues.extend(self._validate_bidirectional_connection(context))
+
+        return issues
+
+    def _validate_bidirectional_connection(
+        self, ctx: ReverseConnectionContext
+    ) -> list[ConsistencyError] | list[Recommendation]:
+        """Validate a single bidirectional connection pair.
+
+        Args:
+            ctx: Connection context containing all necessary references
+
+        Returns:
+            List of validation issues found
+        """
+
+        # Validate source view exists
+        source_view = self._select_source_view(ctx.source_view_ref, ctx.through)
+        if error := self._check_source_view_exists(source_view, ctx):
+            return [error]
+
+        source_view: ViewRequest  # for type checker, since we checked for None above
+        # Validate source property exists and is correct type
+        if error := self._check_source_property(source_view, ctx):
+            return [error]
+
+        source_property = cast(ViewCorePropertyRequest, source_view.properties[ctx.through.identifier])
+
+        # Validate container mapping
+        if container_errors := self._validate_container_mapping(source_property, ctx):
+            return container_errors
+
+        # Validate target view reference
+        return self._validate_target_reference(source_property, ctx)
+
+    def _normalize_through_reference(
+        self, source_view_ref: ViewReference, through: ContainerDirectReference | ViewDirectReference
+    ) -> ViewDirectReference:
```

Comment:
I am a bit surprised that you normalize to ViewDirectReference and not to ContainerDirectReference.